framework/mesh/mesh_generator/from_file_mesh_generator.cc

```diff
@@ -62,31 +58,26 @@ FromFileMeshGenerator::GenerateUnpartitionedMesh(std::unique_ptr<UnpartitionedMe
   options.boundary_id_fieldname = boundary_id_fieldname_;
 
   const std::filesystem::path filepath(filename_);
-  const std::string extension = filepath.extension();
-
-  auto umesh = std::make_unique<UnpartitionedMesh>();
+  AssertReadableFile(filename_);
 
   log.Log() << "FromFileMeshGenerator: Generating UnpartitionedMesh";
-
+  const std::string extension = filepath.extension();
   if (extension == ".obj")
-    umesh->ReadFromWavefrontOBJ(options);
+    return ObjIO::FromFile(options);
   else if (extension == ".msh")
-    umesh->ReadFromMsh(options);
+    return GmshIO::FromFile(options);
   else if (extension == ".e")
-    umesh->ReadFromExodus(options);
+    return VtkIO::FromExodus(options);
   else if (extension == ".vtu")
-    umesh->ReadFromVTU(options);
+    return VtkIO::FromVTU(options);
   else if (extension == ".pvtu")
-    umesh->ReadFromPVTU(options);
+    return VtkIO::FromPVTU(options);
   else if (extension == ".case")
-    umesh->ReadFromEnsightGold(options);
+    return VtkIO::FromEnsightGold(options);
```

Comment:
Would it make more sense to have `ObjIO::FromOBJ` and `GmshIO::FromGmsh` instead to be more consistent with routines in `VtkIO`?

Reply:
Also, should we change `VtkIO` to `VTKIO` since VTK is an all caps acronym?

Reply:
> Would it make more sense to have ObjIO::FromOBJ and GmshIO::FromGmsh instead to be more consistent with routines in VtkIO?

Or, I could do it the other way around and have `VtkIO::FromFile()` and do the split on the type inside 🤔 There is quite a bit of overlap inside the VtkIO class that I am not dealing with in the PR, but it is something that deserves attention.

Reply:
`FromOBJ` should have been `FromObj`. What are your thoughts on my proposal from my comment?

> Having something like MeshIO, XSIO, FieldFunctionIO classes that contain all supported filetypes sounds appealing to me. This ties into my other comment about FromFile. In the classes mentioned above, we would have routines like To(From)<file type>.

Reply:
> Having something like MeshIO, XSIO, FieldFunctionIO classes...

This works too. I am assuming, the code would go into `io` dir in the appropriate location, e.g. `framework/mesh/io`. I am thinking one header file, but multiple `.cc` file, per file format...

Also, if `OBJ` becomes `Obj`, then `VTU` and `PVTU` should be `Vtu` and `Pvtu`? Or do you see that as an acronym, thus keep it all-caps?

Reply:
> This works too. I am assuming, the code would go into io dir in the appropriate location, e.g. framework/mesh/io. I am thinking one header file, but multiple .cc file, per file format...

I would say either multiple `.cc` files or a single file `mesh_io.h(cc)`. 

> Also, if OBJ becomes Obj, then VTU and PVTU should be Vtu and Pvtu? Or do you see that as an acronym, thus keep it all-caps?

I would try to name everything as it is commonly written. If `OBJ` is an acronym, then it should be all caps if that is how it is normally written. I don't lean heavily one way or the other, I just think it is good to establish a standard for it.

Reply:
FWIIW, on Wikipedia, it's VTK (hence I assume also VTU) and OBJ.